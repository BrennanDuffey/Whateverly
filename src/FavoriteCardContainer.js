import React from 'react';
import AnimalCard from './AnimalCard.js';

const FavoriteCardContainer = (props) => {
  return (
    <section> 
      {/* <h2>Favorites</h2> */}
      {props.endangeredSpecies.filter(species => props.favAnimals.includes(species.name))
      .map(species => {
          return <AnimalCard {...species} key={species.name} addFav={props.addFav} favAnimals={props.favAnimals}/>
      })}
    </section>
  )
}

export default FavoriteCardContainer;

