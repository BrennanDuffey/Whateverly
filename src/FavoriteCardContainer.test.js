import React from 'react';
import FavoriteCardContainer from './FavoriteCardContainer.js';
import { shallow } from 'enzyme';

const mockEndageredSpecies = [
  {
    "name": "Snow Leopard",
    "genus": "Panthera",
    "endangeredStatus": "Vulnerable",
    "locations": ["China", "Mongolia", "India"],
    "population": 6000,
    "threats": ["Poaching","Climate Change"],
    "image": "https://upload.wikimedia.org/wikipedia/commons/d/d7/Snow_Leopard_in_Ladakh%28_Photo_by_Tashi_Lonchay%29.jpg"
  },
  {
    "name": "Jaguar",
    "genus": "Panthera",
    "endangeredStatus": "Near Threatened",
    "locations": ["Amazonian Brazil", "South America", "Central America", "Mexico" ],
    "population": "Unknown",
    "threats": ["Poaching", "Habitat Loss", "Human Intolerance" ],
    "image": "https://images.unsplash.com/photo-1517825738774-7de9363ef735?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
  }
];
const mockFavAnimals = ["Snow Leopard"];
const mockAddFav = jest.fn();

describe('FavoriteCardContainer', () => {
  
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <FavoriteCardContainer endangeredSpecies={mockEndageredSpecies} 
      addFav={mockAddFav} 
      favAnimals={mockFavAnimals}/>
    )
  });

  it('Should render correctly with all the information passed', () => {
      expect(wrapper).toMatchSnapshot();
  });

  
});