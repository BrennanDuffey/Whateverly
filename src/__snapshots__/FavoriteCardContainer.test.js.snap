// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCardContainer Should render correctly with all the information passed 1`] = `
<section>
  <AnimalCard
    addFav={[MockFunction]}
    endangeredStatus="Vulnerable"
    favAnimals={
      Array [
        "Snow Leopard",
      ]
    }
    genus="Panthera"
    image="https://upload.wikimedia.org/wikipedia/commons/d/d7/Snow_Leopard_in_Ladakh%28_Photo_by_Tashi_Lonchay%29.jpg"
    key="Snow Leopard"
    locations={
      Array [
        "China",
        "Mongolia",
        "India",
      ]
    }
    name="Snow Leopard"
    population={6000}
    threats={
      Array [
        "Poaching",
        "Climate Change",
      ]
    }
  />
</section>
`;
